src/utils.js

```diff
@@ -58,6 +58,33 @@ export function getRootExpression(node) {
 }
 
 /*
+ * returns the name of the reference node passed.
+ *
+ * example: var foo = document;
+ *  The node for foo will return 'document'
+ */
+export function getNodeReferenceName(context, node) {
+  var variables = context.getScope().variables;
+  var scopeVar;
+
+  variables.forEach((variable) => {
+    if (variable.name === node.name) {
+      scopeVar = variable;
+    }
+  });
+
+  if (typeof scopeVar === 'undefined' ||
+      typeof scopeVar.defs === 'undefined' ||
+      typeof scopeVar.defs[0] === 'undefined' ||
```

Comment:
Do further list items in defs mean anything in this context? Definitely would help to have some comments so that looking at this code it's clearer without needing to look up eslint docs.